controllers/clustergroupupgrade_controller.go

```diff
@@ -1421,45 +1421,25 @@ func (r *ClusterGroupUpgradeReconciler) getPolicyClusterStatus(policy *unstructu
 
 func (r *ClusterGroupUpgradeReconciler) getClustersNonCompliantWithPolicy(
 	ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade,
-	policy *unstructured.Unstructured, filterClustersForUpgrade bool) ([]string, error) {
-
-	// Get the status of the clusters matching the policy.
-	subStatus, err := r.getPolicyClusterStatus(policy)
-	if err != nil {
-		return nil, err
-	}
+	policy *unstructured.Unstructured) ([]string, error) {
 
 	var nonCompliantClusters []string
-	// Loop through all the clusters in the policy's compliance status.
-	for _, crtSubStatusCrt := range subStatus {
-		crtSubStatusMap := crtSubStatusCrt.(map[string]interface{})
-		// If the cluster is NonCompliant, add it to the list. A cluster without a status is also considered NonCompliant.
-		if crtSubStatusMap["compliant"] == utils.ClusterStatusNonCompliant || crtSubStatusMap["compliant"] == nil {
-			nonCompliantClusters = append(nonCompliantClusters, crtSubStatusMap["clustername"].(string))
-		}
+	var nonCompliantClustersInUpgrade []string
+	allClustersForUpgrade, err := r.getAllClustersForUpgrade(ctx, clusterGroupUpgrade)
+	if err != nil {
+		return nil, fmt.Errorf("cannot obtain all the details about the clusters in the CR: %s", err)
 	}
-
-	// Filter only the clusters present in the current upgrade.
-	if filterClustersForUpgrade {
-		var nonCompliantClustersInUpgrade []string
-		allClustersForUpgrade, err := r.getAllClustersForUpgrade(ctx, clusterGroupUpgrade)
+	for _, cluster := range allClustersForUpgrade {
+		compliance, err := r.getClusterComplianceWithPolicy(cluster, policy)
```

Comment:
we can't redo this for every single cluster. it's the same api call.

Reply:
That makes sense however I kinda repeated what `getClustersNonCompliantWithManagedPolicies` does :laughing:

Reply:
I am on it as soon as it passes the tests

Reply:
Don't worry about it in this PR then. Let's get this fix in first.